config.md

```diff
@@ -399,6 +403,18 @@ The [state](runtime.md#state) of the container MUST be passed to hooks over stdi
 The pre-start hooks MUST be called after the [`start`](runtime.md#start) operation is called but [before the user-specified program command is executed](runtime.md#lifecycle).
 On Linux, for example, they are called after the container namespaces are created, so they provide an opportunity to customize the container (e.g. the network namespace could be specified in this hook).
 
+### <a name="configHooksMount" />Mount Hooks
+
+The mount hooks MUST be called as part of the create operation at a time that would allow the hook to execute operations both inside and outside the container.
+For example, on linux this would happen before the `pivot_root` operation is executed but after the mount namespace was created and setup.
+
+For runtimes that implement the deprecated pre-start hooks as mount hooks, mount hooks MUST be called after the prestart hooks.
```

Comment:
I'm not sure we should specify the order. Since the definition of the prestart hook is not changed, this order makes it impossible to implement a container runtime with support for the two hooks following the spec. And for compatibility, hooks installed on both prestart and mount could both check if the mount task is already done, so the order should not matter.

Reply:
> And for compatibility, hooks installed on both prestart and mount could both check if the mount task is already done, so the order should not matter.

 The goal here was that you can rely on ordering instead of checking for state you might have persisted in the mount or prestart hook.

> this order makes it impossible to implement a container runtime with support for the two hooks following the spec

Do you mind detailling a bit more what you mean here :) ?

Reply:
> Do you mind detailing a bit more what you mean here :) ?

Sorry, I take it back.

So now, the spec mandates steps to happen in this order:
1. the container namespaces are created
2. Prestart hooks
3. Mount hooks
4. pivot_root
5. Start hooks
6. The user-specified process is executed 
7. Poststart hooks

I thought the spec was saying that the Prestart hooks must happen after the namespaces are fully configured (including the pivot_root), which would make things impossible. But the wording is more vague:

> [The pre-start hooks] are called after the container namespaces are created

So I think it's ok :)